File: 643_Maximum_Average_Subarray_I/solution.c
double findMaxAverage(int* nums, int numsSize, int k) {
    long sum = 0, max = 0;
    for (int i = 0; i < k; i++) sum += nums[i];
    max = sum;

    for (int i = k; i < numsSize; i++) { sum = sum - nums[i - k] + nums[i]; max = (sum > max) ? sum : max; }
    return (double)max / k;
}

File: 643_Maximum_Average_Subarray_I/solution.py
class Solution:
    def findMaxAverage(self, nums: List[int], k: int) -> float:
        window_sum = sum(nums[:k])
        maximum_sum = window_sum

        for i in range(k, len(nums)):
            window_sum = window_sum - nums[i - k] + nums[i]
            maximum_sum = max(maximum_sum, window_sum)
        return maximum_sum / k
